src/test/java/bit/javaoop/FixedBudgetMangerTest.java

```diff
@@ -0,0 +1,69 @@
+package bit.javaoop;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.LinkedList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class FixedBudgetMangerTest {
+
+    @Test
+    void canHireEmployeeTest() {
+        Manager manager1 = new FixedBudgetManager("Steve Jobs", 30_000, 800_000);
+        manager1.hireEmployee(new Employee("Garek Majęcki", 60_000));
+
+        Manager manager2 = new FixedBudgetManager("Bill Gates", 30_000, 80_000);
+        manager2.hireEmployee(new Employee("Linus Torvalds", 0)); //He is an open source guy.
+
+        boolean result1 = manager1.canHireEmployee(new Employee("Alan Turing", 100_000));
+        boolean result2 = manager2.canHireEmployee(new Employee("Alan Turing", 100_000));
+
+        assertTrue(result1);
+        assertFalse(result2);
+    }
+
+    @Test
+    void hireEmployeeTest() {
+
+        Manager fixedBudgetManager = new FixedBudgetManager("Steve Jobs", 30_000, 80_000);
+        Employee bestEmployee = new Employee("Garek Majęcki", 60_000);
+
+        fixedBudgetManager.hireEmployee(bestEmployee);
+        LinkedList<Employee> employees = fixedBudgetManager.getEmployeeList();
+
+        assertEquals(bestEmployee, employees.getFirst());
+    }
+
+    @Test
+    void isSatisfiedTest() {
```

Comment:
Widzę tutaj konwencję pisania testów: //given //when //then, przedstawioną na zajęciach. Wg mnie fajnie jest się do niej stosować, bo na dłuższą metę bardzo "uprzejrzyścia" kod. Jedyne co musiałbyś dodać to komentarze: właśnie //given //when oraz //then. Czyli tutaj byłoby mniej więcej coś takiego:
```
//given
Manager manager1 = new FixedBudgetManager("Steve Jobs", 30_000, 80_000);
//when
manager1.hireEmployee(new Employee("Garek Majęcki", 60_000));
//then
assertTrue(manager1.isSatisfied());
```

Nazwy metod mogłyby być inne. Intellij sugeruje takie jak masz, ale lepiej trzymać się konwencji `shouldDoSomething`, czyli np. w naszym przypadku `shouldBeSatisfied`.

Dodatkowo powiem, że wyżej specjalnie napisałem jeden rodzaj assertu, gdyż możnaby rozdzielić ten test na dwa: `shouldBeSatisfied` oraz `shouldNotBeSatisfied`. Taki podział na "klasy abstrakcji" również powinno się stosować.

PS. Plus za fajne imiona :)

Reply:
Wszedzie miałem tak:
//given
//when
///then
Ale usunalem bo myslałem ze to nieprofesjonalne :/

Reply:
Jak dla większości przypadków komentarze są nieprofesjonalne, tak w tym wypadku akurat są zalecane. Ułatwiają potem czytanie kodu. Dokładnie wiadomo CO jest testowane (blok //when) po jednym zerknięciu okiem.